plugins/Xpressive/Xpressive.cpp

```diff
@@ -572,20 +572,21 @@ void XpressiveView::expressionChanged() {
 			expr.add_cyclic_vector("W2",e->graphW2().samples(),e->graphW2().length());
 			expr.add_cyclic_vector("W3",e->graphW3().samples(),e->graphW3().length());
 		}
-		expr.setIntegrate(&i,sample_rate);
+        expr.setIntegrate(&frame_counter,sample_rate);
 		expr.add_constant("srate",sample_rate);
 
 		const bool parse_ok=expr.compile();
 
 		if (parse_ok) {
 			e->exprValid().setValue(0);
-			const auto length = static_cast<std::size_t>(m_raw_graph->length());
+            const unsigned int length = static_cast<unsigned int>(m_raw_graph->length());
 			auto const samples = new float[length];
-			for (auto i = std::size_t{0}; i < length; i++) {
-				t = i / (float) length;
-				samples[i] = expr.evaluate();
-				if (std::isinf(samples[i]) != 0 || std::isnan(samples[i]) != 0)
-					samples[i] = 0;
+            // frame_counter's reference is used in the integrate function.
+            for ( frame_counter = 0; frame_counter < length; ++frame_counter) {
+                t = frame_counter / (float) length;
```

Comment:
You have changed this line, so better to use `static_cast` per our guidelines. 

```suggestion
                t = frame_counter / static_cast<float>(length);
```

Reply:
Btw, `t` looks like an unused variable. Might as well check that one.

Reply:
No. The expressions evaluator has a reference to the variable t, and it is essential to update t.

Reply:
Ohh i see.